### PR TITLE
Adding more logic to pick a better package before picking up the first package in the list of available packages

### PR DESCRIPTION
**Pull request by brdandu**
…

### CODE REVIEW

**Review by tecimovic — APPROVED** [review]
We can do this as a temporary workaround.

But we REALLY have to weed out all the code that EVER "picks the first one", because it doesn't know what to do.

If you can't make an unambiguous decision, you NEED TO defer it back to the user.

File: src-electron/importexport/import-json.js [hunk 1]
```diff
@@ -213,11 +213,30 @@ async function importSinglePackage(db, pkg, zapFilePath, packageMatch) {
...
   } else {
-    // None exists, so use the first one from 'packages'.
     let p = packages[0]
-    env.logWarning(
-      `None of packages exist, so using first one overall: ${p.id}.`
+    let pkgPaths = packages.map((p) => p.path)
+    //eslint-disable-next-line
+    let filePathToSearch = pkg.path.split(/.*[\/|\\]/).pop()
+    let packageNameMatch = packages.find(
+      (p) => p.path.includes(filePathToSearch) && fs.existsSync(p.path)
```
Comment (tecimovic) [inline src-electron/importexport/import-json.js:221]: Generally, we should never use `.existsSync`.

**Comment by brdandu** [comment]
> We can do this as a temporary workaround.
> 
> But we REALLY have to weed out all the code that EVER "picks the first one", because it doesn't know what to do.
> 
> If you can't make an unambiguous decision, you NEED TO defer it back to the user.

Added more logic to this such that the passed zcl and temaplates json files are used when generate is called directly without opening up the UI

**Review by paulr34 — COMMENTED** [review]
What does this solve?

**Comment by paulr34** [comment]
> We can do this as a temporary workaround.
> 
> 
> 
> But we REALLY have to weed out all the code that EVER "picks the first one", because it doesn't know what to do.
> 
> 
> 
> If you can't make an unambiguous decision, you NEED TO defer it back to the user.

@tecimovic it doesnt "pick the first one". It simply passes in 1 file when the database is broken.

**Pull request merged**